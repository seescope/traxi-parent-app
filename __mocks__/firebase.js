export const profile = {
  UUID: 'YwS0vJ8OE8N6yenxHaV6PdMVLbG3',
  email: 'enakudesu@gmail.com',
  kids: {
    '0': {
      UUID: '2f566920-f598-46d2-8bf2-7bcae115bf0a',
      avatarURL: 'http://i.imgur.com/52mRwuE.jpg',
      deviceType: 'Android',
      name: 'Sam',
      setupID: 5406,
      status: 'INSTALLED',
    },
    '1': {
      UUID: '886b79ea-3572-4b4a-9c25-59b14639ac3d',
      avatarURL: 'https://tiestoclublife.files.wordpress.com/2008/08/dj-tiesto-club-life.jpeg',
      name: 'DJ',
    },
  },
  name: 'Kane Rogers',
  picture: 'https://lh3.googleusercontent.com/-7yt2pBdbMwk/AAAAAAAAAAI/AAAAAAAAD0g/4IHp0iIWDBE/s96-c/photo.jpg',
  token: 'cjj6guKMTME:APA91bFkM3nynjnD5vqxxeYG42t4P6PThP4N1F7VmhNbRW_W_ReCqGtoiDtwG99K4GmD6_Sh_VuCFCR46GwXH314YfdciS0OTGdwe0OGxYvT2RiYhozwcmlLI2TyHn78z26Kw7SYquCD',
};

export const initializeApp = () => {};
export const database = () => ({
  ref: () => ({
    once: () =>
      Promise.resolve({
        val: () => profile,
      }),
  }),
  goOffline: () => {},
});

const firebase = {};

export default firebase;
