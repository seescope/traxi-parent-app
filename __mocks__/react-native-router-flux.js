import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
