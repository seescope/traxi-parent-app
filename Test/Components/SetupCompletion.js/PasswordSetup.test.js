import * as Firebase from 'firebase';
import { Alert, AsyncStorage } from 'react-native';
import { Actions } from 'react-native-router-flux';
import React from 'react';
import renderer from 'react-test-renderer';
import {
  PasswordSetup,
  alertIfPasswordIsTooShort,
  updateFirebaseProfile,
  createFirebaseUser,
  saveProfileToAsyncStorage,
  goToDashboard,
} from '../../../App/Components/SetupCompletion/PasswordSetup';

Alert.alert = jest.fn();
Actions.dashboard = jest.fn();

describe('NameSetup', () => {

  it('Renders correctly', () => {
    const setEmailFn = jest.fn();
    const name = 'Chris';
    const email = 'test@email.com';

    const tree = renderer.create(
      <PasswordSetup name={name} email={email} setEmailFn={setEmailFn} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('alertIfPasswordIsTooShort()', () => {
    expect(alertIfPasswordIsTooShort('password')).toBeFalsy();
    expect(Alert.alert).not.toHaveBeenCalled();
    expect(alertIfPasswordIsTooShort('short')).toBeTruthy();
    expect(Alert.alert).toHaveBeenCalledWith(
      'Password should be at least 6 characters',
    );
  });

  it('updateFirebaseProfile()', () => {
    //TODO
  });

  it('saveProfileToAsyncStorage()', () => {
    saveProfileToAsyncStorage('UUID');
    expect(AsyncStorage.setItem).toHaveBeenCalledWith(
      'profile',
      JSON.stringify({
        UUID: 'UUID',
      }),
    );
  });

  it('createFirebaseUser()', () => {
    //TODO
  });

  it('goToDashboard()', () => {
    goToDashboard();
    expect(Actions.dashboard).toHaveBeenCalled();
  });
});
