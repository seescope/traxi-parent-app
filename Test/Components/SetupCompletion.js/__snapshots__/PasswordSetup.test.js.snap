// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSetup Renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#448AFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 6,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 30,
          "marginBottom": -2,
          "textAlign": "center",
        }
      }
    >
      Last step, 
      Chris
       !
    </Text>
    <View
      style={
        Object {
          "height": 32,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Raleway-Regular",
          "fontSize": 16,
        }
      }
    >
      Your email address:
    </Text>
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={undefined}
      placeholderTextColor="rgba(75, 73, 73, 0.8)"
      secureTextEntry={undefined}
      style={
        Object {
          "backgroundColor": "#EFEFEF",
          "borderRadius": 4,
          "color": "rgba(75, 73, 73, 1)",
          "fontFamily": "Raleway-Regular",
          "height": 32,
          "paddingHorizontal": 8,
          "width": 686,
        }
      }
      underlineColorAndroid="#448AFF"
      value="test@email.com"
    />
    <View
      style={
        Object {
          "height": 32,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontFamily": "Raleway-Regular",
          "fontSize": 16,
        }
      }
    >
      Choose a password:
    </Text>
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType={undefined}
      onChangeText={[Function]}
      onSubmitEditing={undefined}
      placeholderTextColor="rgba(75, 73, 73, 0.8)"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#EFEFEF",
          "borderRadius": 4,
          "color": "rgba(75, 73, 73, 1)",
          "fontFamily": "Raleway-Regular",
          "height": 32,
          "paddingHorizontal": 8,
          "width": 686,
        }
      }
      underlineColorAndroid="#448AFF"
      value={undefined}
    />
    <View
      style={
        Object {
          "height": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }
      >
        Finished
         
      </Text>
    </View>
  </View>
</View>
`;
